src/Paprika/Store/StateRootPage.cs

```diff
@@ -74,7 +74,27 @@ public Page DeleteByPrefix(in NibblePath prefix, IBatchContext batch)
             return new StateRootPage(writable).DeleteByPrefix(prefix, batch);
         }
 
-        Debug.Assert(prefix.Length > ConsumedNibbles, "Removing prefix at the root? Something went wrong.");
+        if (prefix.Length < ConsumedNibbles)
+        {
+            if (prefix.Length == 1)
+            {
+                var idx = prefix.Nibble0 << NibblePath.NibbleShift;
+                for (int i = 0; i < 16; i++)
+                {
+                    ref var addrShort = ref DeleteAll(Data.Buckets, idx + i);
+                }
+            }
+            else if (prefix.IsEmpty)
+            {
+                //can all pages be freed here?
```

Comment:
I think they can but this requires a separate PR. I think it should be up to the pages to register themselves and return a `null page` there.